### ISSUE TITLE
Natlink Unable to Open Config GUI

### ISSUE THREAD

**Issue by alexrkoch** [dictation-toolbox/natlink#19]
Hi- I'm a first time NatLink user doing my initial installation, following [this page ] (https://qh.antenna.nl/unimacro/installation/installationstableversionpython2.html). Here is a history of the steps taken for context:
1) Installed Dragon Home 15.3
2) Installed Natlink, then uninstalled after reading that I need Python 2.7 first (I have 3.7 installed currently).
3) Install Python 2.7.16 (32bit) directly from python.org
4) pip install: wxpython, pywin32, six, future.

ISSUE: When I try: Start > Configure NatLink via GUI, I get this error message:

```
Traceback (most recent call last):
  File "C:\Natlink\Natlink\confignatlinkvocolaunimacro\configurenatlink.pyw", line 1363, in <module>
    app = MyApp(True)
  File "C:\Python27\lib\site-packages\wx\core.py", line 2194, in __init__
    self._BootstrapApp()
SystemExit: OnInit returned false, exiting...

```
I've referenced this to @lexxish comment below because its the same error (with just slightly different file path), however there was no resolution on that thread.

PS- this is also my first GitHub post, so if you have any feedback on how to improve clarity, etc., I would be appreciative.

Many thanks! 
Alex

> The python 3 is still work in progress and unfortunately we have not (and might not be able to) create a stable ABI for python 3.X. Currently I am compiling to python 3.7 and @mrob95 is compiling to 3.8.
> 
> The goal is that it will work seem less with both versions eventually. Thanks for posting the work around.

I wanted to try python 3.6 because [psychopy ](https://www.psychopy.org/api/iohub/device/eyetracker_interface/Tobii_Implementation_Notes.html) seems to work only with 3.6 but getting the below error. Any thoughts on why? I was thinking we could add some eye tracking features directly into caster with psychopy.

```
Traceback (most recent call last):
  File "configurenatlink.pyw", line 1363, in <module>
    app = MyApp(True)
  File "C:\Users\user\AppData\Local\Programs\Python\Python36\lib\site-packages\wx\core.py", line 2194, in __init__
    self._BootstrapApp()
SystemExit: OnInit returned false, exiting...

```

_Originally posted by @lexxish in https://github.com/dictation-toolbox/natlink/issues/2#issuecomment-619314259_

**Comment by kb100** [comment]
I had this same issue when trying to set up natlink, which is why I created my fork of natlink, which supports python 3.6, 3.7, and 3.8. It is not "stable", but I would appreciate any feedback from users willing to try it.

**Comment by LexiconCode** [comment]
I believe this issue stems from `wxpython 4.1.0` try `python -m pip install wxpython==4.0.7`

**Comment by alexrkoch** [comment]
> I believe this issue stems from `wxpython 4.1.0` try `python -m pip install wxpython==4.0.7`

I installed wxpython 4.0.7, rebooted, and got the same error still. Should I try uninstalling then re-installing Natlink after having changed wxpython? Or any other additional steps?



> I had this same issue when trying to set up natlink, which is why I created my fork of natlink, which supports python 3.6, 3.7, and 3.8. It is not "stable", but I would appreciate any feedback from users willing to try it.

If there are no other suggested solutions to the issue I will give that a try and collect feedback 👍🏻

**Comment by LexiconCode** [comment]
> 
> 
> > I believe this issue stems from `wxpython 4.1.0` try `python -m pip install wxpython==4.0.7`
> 
> I installed wxpython 4.0.7, rebooted, and got the same error still. Should I try uninstalling then re-installing Natlink after having changed wxpython? Or any other additional steps?

I can't think of any additional steps but others might have some suggestions.

> If there are no other suggested solutions to the issue I will give that a try and collect feedback 👍🏻

To my knowledge kb100 fork is going to be merged upstream as soon as a few other bits are tweaked in the main repository. So I would recommend giving his fork a try.

> [psychopy ](https://www.psychopy.org/api/iohub/device/eyetracker_interface/Tobii_Implementation_Notes.html) seems to work only with 3.6 but getting the below error. Any thoughts on why? I was thinking we could add some eye tracking features directly into caster with psychopy

You're welcome to try although 3.6 is pretty limiting. I was thinking of moving along with Caster with Python 3 starting at v3.7-3.8.
I was thinking of actually integrating the following [OCR and Eye Tracking libraries](https://handsfreecoding.org/2020/07/25/say-what-you-see-efficient-ui-interaction-with-ocr-and-gaze-tracking/) see the link for demo and explanation. Once a PR is merged into dragonfly dealing with Windows DPI scaling this will be integrated into Caster.

It may be possible use [stream engine](https://developer.tobii.com/consumer-eye-trackers/stream-engine/getting-started/) and load the appropriate dlls/api through Python. If that is the case it will be very similar to what's implemented for Windows. From there we can work on Linux support to its underlying libraries [gaze-ocr](https://github.com/wolfmanstout/gaze-ocr/) and [screen-ocr](https://github.com/wolfmanstout/screen-ocr)

**Comment by alexrkoch** [comment]
Okay thanks for your help, I will try kb100's fork.

**Comment by alexrkoch** [comment]
FYI, original issue fully resolved-  I was able to get the python 2.7 version of Natlink configure GUI working by following @LexiconCode advice, except using ```wxpython 4.0.3```  rather than version 4.0.7

**Comment by DanKaplanSES** [comment]
Can this be reopened so this information is added to the readme?

 Brand new windows 10 install. I needed to take these steps:

```powershell
gcm python # this should output a path to Python 2.7
gcm pip # this should output a path to Python 2.7

cd 'C:\Natlink\Natlink\confignatlinkvocolaunimacro'
pip install pywin32
pip install setuptools # this may not be necessary. I uninstalled it as part of my troubleshooting steps
pip install wxPython==4.0.3
python start_configurenatlink.py
```

**Comment by LexiconCode** [comment]
I don't think they're supporting python 2.7 any more.
